internal/services/containers/kubernetes_nodepool.go

```diff
@@ -292,6 +291,11 @@ func SchemaDefaultNodePool() *pluginsdk.Schema {
 				}
 
 				if !features.FourPointOhBeta() {
+					s["type"].ValidateFunc = validation.StringInSlice([]string{
```

Comment:
Unless there are plans by the service team to introduce a new type for clusters to run on, the `type` property should be deprecated and removed in 4.0 with the provider defaulting this to VMSS for the user.

Reply:
@hqhqhqhqhqhqhqhqhqhqhq any update in this? This should go in soon.

Reply:
Waiting for updates from service team.
I just made a commit for removing `Type` (still left as computed to be read out) for review purposes.
If there is plan from service team to add more types in future, I will revert the commit.

test case:
![img_v3_02di_059c011a-3da3-42ae-bd85-df531164c5hu](https://github.com/user-attachments/assets/1354d8cd-037b-48b1-944a-7bce61f87d84)

Reply:
@stephybun 
Service team indicated that there will new type coming soon. I will revert the commit and keep the `Type` property.

Reply:
Also the new type would also accept Node Image auto upgrade